packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -0,0 +1,89 @@
+import { Payment } from "@models/payment"
+import { Cart } from "@models/cart"
+import { PaymentSession, PaymentSessionStatus } from "@models/payment-session"
+import { Customer } from "@models/customer"
+import { BaseService } from "medusa-interfaces"
+
+export type Data = Record<string, unknown>
+export type PaymentData = Data
+export type PaymentSessionData = Data
+
+export interface PaymentService {
+  getIdentifier(): string
+
+  onCreatePaymentSession(cart: Cart): Promise<PaymentSessionData>
+
+  onCreatePayment(paymentSession: PaymentSession): Promise<PaymentData>
+
+  retrievePayment(cart: Cart): Promise<Payment>
+
+  updatePayment(
```

Comment:
`onUpdatePayment`

Reply:
This one would be more `onUpdateSession`